packages/block-library/src/navigation/constants.js

```diff
@@ -5,6 +5,7 @@ export const DEFAULT_BLOCK = {
 export const ALLOWED_BLOCKS = [
 	'core/navigation-link',
 	'core/search',
```

Comment:
Would match the current behavior more closely `core/search` was not allowed, and only the inline variation is?